src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -566,6 +569,11 @@ class RegularSync(
   private def notDownloading(): Boolean =
     headersQueue.isEmpty && waitingForActor.isEmpty && !resolvingBranches
 
+  private def publishEvents(newBlocks: Seq[Block]): Unit = {
+    newBlocks.foreach { newBlock =>
+      context.system.eventStream.publish(NewHead(newBlock))
```

Comment:
Do we need to publish it for all blocks and not just the latest one?

Reply:
For all blocks